### PR TITLE
[Keymap] Add brauner preonic layout

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <brauner@kernel.org>

Add a new keymap for the preonic keyboard.

## Description

This is a keymap layout heavily optimized for vim.
It mostly follows `QWERTY` but moves various symbols on to different layers.

* Number keys function only as number keys.
  Additional symbols they would produce when `Shift` is held are turned off as they are intercepted.
* `Y`, `P`, and `X` function as `Ctrl+C`, `Ctrl+V`, and `Ctrl+X` when held.
  This avoids having to hold down `Ctrl` and also aligns regular copy and paste with vim functionality.
* Various keys perform different functions when tapped than when held.
* `!` and `?` are on top of `.` on different layers.
* `\` and `|` are on top of `/` on different layers.
* `:` and `;` are on top of `,` on different layers.
* All navigation is done via `H`, `J`, `K`, and `L`.
* `O`, `U`, `S`, and `A` produce the Umlauts `Ö`, `Ü`, `ß`, and `Ä` when held.
  This relies on the operating system keyboard layout to be set to US Intl.
* Different layers will activate different types of navigation.
  For example, left, right, up, and down or backspace, page up, page down, and tab.
* There's a dedicated mouse layer which can be activated allowing to navigate solely with the keyboard where otherwise a mouse or trackpad would be needed.
* There's a dedicated management layer where the boot loader can be entered or music mode can be turned on or off.
* Lastly, there's a compositor layer which is optimized for managing a tiling window manager.
  The layer is used specifically via layer-mod keys in the base layer.
  In other words, specific modifiers are held when this layer is accessed.
* Opening and closing brackets are always located on adjacent keys on the same layer.
  Pressing a matching opening and closing bracket key at the same time will insert both brackets and move the cursor into the middle of the bracket pair.
  As should be obvious, this is useful when programming: `(|)`, `[|]`, `{|}`, `<|>`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by brauner** [comment]
So I've looked through git history and I'm a bit puzzled that so many new keymap additions have a `Co-developed-by:` tag attached to them. As a kernel maintainer and developer that tag is really only given when most of the contents of the patch are based on the joint work of multiple people.

In any case, happy to hear your thoughts. I've squashed all commits I had into a single one and tried to follow the example of the other keymap additions.

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (drashna) [inline keyboards/preonic/keymaps/brauner/keymap.c:341]: you could use `tap_code` for these.  Simpler, and handles waits between, if enabled.

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (drashna) [inline keyboards/preonic/keymaps/brauner/keymap.c:376]: You could replace this with `tap_code16(RCTL(keycode))`

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (drashna) [inline keyboards/preonic/keymaps/brauner/keymap.c:402]:
```suggestion
            tap_code16(C(S(KC_C)));
```

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (drashna) [inline keyboards/preonic/keymaps/brauner/keymap.c:412]:
```suggestion
            tap_code16(C(S(KC_V)));
```

**Review by drashna — COMMENTED** [review]
Mostly, just some clean up suggestions

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (brauner) [reply keyboards/preonic/keymaps/brauner/keymap.c:341]: done

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (brauner) [reply keyboards/preonic/keymaps/brauner/keymap.c:376]: done

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (brauner) [reply keyboards/preonic/keymaps/brauner/keymap.c:402]: done

File: keyboards/preonic/keymaps/brauner/keymap.c
Comment (brauner) [reply keyboards/preonic/keymaps/brauner/keymap.c:412]: done

**Comment by brauner** [comment]
> Mostly, just some clean up suggestions

Thanks! I incorporated them and force-pushed a new version just now.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by brauner** [comment]
Seems that this is ready to merge? Let me know if there's anything else I need to do.

**Comment by brauner** [comment]
Thank you!

**Pull request merged**
